provisioning/roles/wg/tasks/main.yml

```diff
@@ -0,0 +1,93 @@
+ ---
+- name: Add debian unstable repo
+  copy:
+    src: templates/unstable.list
+    dest: /etc/apt/sources.list.d/unstable.list
+    owner: root
+    group: root
+    mode: '0644'
+
+- name: Pin unstable repo
+  copy:
+    src: templates/limit-unstable
+    dest: /etc/apt/preferences.d/limit-unstable
+    owner: root
+    group: root
+    mode: '0644'
+
+- name: Install wireguard
+  apt:
+    update_cache: yes
+    state: present
+    name: wireguard
+
+- name: Read private client's key
+  stat:
+    path: "/etc/wireguard/privkey"
+  register: privkey
+
+- name: Generate wireguard client's keys
+  shell: set -o pipefail && \
+         umask 077; wg genkey | tee /etc/wireguard/privkey | wg pubkey > /etc/wireguard/pubkey
+  args:
+    executable: /bin/bash
+  when:
+    - not privkey.stat.exists
+
+- name: Read privkey
+  tags: genConfig
+  slurp:
+    src: "/etc/wireguard/privkey"
+  register: privkey
+
+- name: Read pubkey
+  slurp:
+    src: "/etc/wireguard/pubkey"
+  register: pubkey
+
+- name: show pubkey
```

Comment:
It would be better to save this to a dict of hostname:pubkey and print them all at the end of the run

Reply:
I've never done that with Ansible, would you mind doing this?